lib/senkyoshi/models/file_resource.rb

```diff
@@ -0,0 +1,23 @@
+require "senkyoshi/models/resource"
+
+module Senkyoshi
+  ##
+  # Class to represent a resource constructed from a single 'dat' file.
+  ##
+  class FileResource < Resource
+    attr_reader(:id)
+
+    def initialize(id = nil)
+      @id = id
+    end
+
+    def self.from(xml = nil, pre_data = nil, _resource_xids = nil)
+      resource = new(pre_data[:file_name])
+      resource.iterate_xml(xml, pre_data)
```

Comment:
Is it helpful to pass `xml` and `pre_data` to this method call when the method just ignores them?